x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/io/stream/PlanNamedTypes.java

```diff
@@ -1485,6 +1489,28 @@ static void writeIntersects(PlanStreamOutput out, SpatialIntersects intersects)
         out.writeExpression(fields.get(1));
     }
 
+    static SpatialContains readContains(PlanStreamInput in) throws IOException {
+        return new SpatialContains(Source.EMPTY, in.readExpression(), in.readExpression());
+    }
+
+    static void writeContains(PlanStreamOutput out, SpatialContains intersects) throws IOException {
+        List<Expression> fields = intersects.children();
+        assert fields.size() == 2;
+        out.writeExpression(fields.get(0));
+        out.writeExpression(fields.get(1));
```

Comment:
```
out.writeExpression(intersects.left())
out.writeExpression(intersects.right())
```
No need to do assertion or generic extractions.

Reply:
Done, and merged the three write functions into one, since they are identical. But we still need three read functions to instantiate the correct classes.